### PR TITLE
Enforce trailing commas in javascript arrays, objects, import, and exports

### PR DESCRIPTION
**Pull request by apmorton**
This is more of an RFC than anything else - just interested in seeing opinions.

The rational here is to make git diffs less noisy when appending something to the end of an array or object literal.

This shouldn't introduce any compatibility issues in browsers, as its been supported for objects/arrays since almost forever:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas

### CODE REVIEW

**Comment by partouf** [comment]
I think I always wondered the linter wouldn't have any of it, because I've seen trailing comma's more often than not. Didn't actually consider why though, but this makes sense. So I'd be ok with this.

File: lib/compilers/ada.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ class AdaCompiler extends BaseCompiler {
                 '-eS', // commands are not errors
                 '-cargs', // Compiler Switches for gcc.
                 '-o',
-                outputFilename,
+                outputFilename
```
Comment (apmorton) [inline lib/compilers/ada.js:58]: just noticed this - so it seems at least _sometimes_ we have functions which use the trailing comma - maybe we should enable this for server side code only?

Browser support is somewhat sketchy, so we couldn't enable it there yet...

Comment (RabsRincon) [reply lib/compilers/ada.js:58]: As per https://eslint.org/docs/rules/comma-dangle#options-14
> functions should only be enabled when linting ECMAScript 2017 or higher.

So we good to enable it server side. I'd say go for it :)

**Comment by apmorton** [comment]
closing in favor of #2121

**Pull request closed without merging**